test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -6132,6 +6132,42 @@ BOOST_AUTO_TEST_CASE(callable_crash)
 	CHECK_ERROR(text, TypeError, "Type is not callable");
 }
 
+BOOST_AUTO_TEST_CASE(warn_transfer_non_payable_fallback)
+{
+	char const* text = R"(
+		contract A {
+			function() {}
+		}
+
+		contract B {
+			A a;
+
+			function() {
+				a.transfer(100);
+			}
+		}
+	)";
+	CHECK_ERROR(text, TypeError, "Value transfer to a contract with a non-payable fallback function.");
+}
+
+BOOST_AUTO_TEST_CASE(warn_send_non_payable_fallback)
+{
+	char const* text = R"(
+		contract A {
+			function() {}
+		}
+
+		contract B {
+			A a;
+
+			function() {
+				if (!a.send(100)) throw;
```

Comment:
Change this to `require(a.send(100))` as `throw` issues a deprecation warning now (and will be removed).